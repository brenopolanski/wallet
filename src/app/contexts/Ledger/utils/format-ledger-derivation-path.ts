import { LedgerDerivationScheme } from "app/contexts/Ledger/contracts";

export const formatLedgerDerivationPath = (scheme: LedgerDerivationScheme) =>
	`m/${scheme.purpose || 44}'/${scheme.coinType}'/${scheme.account || 0}'/${scheme.change || 0}/${
		scheme.address || 0
	}`;
