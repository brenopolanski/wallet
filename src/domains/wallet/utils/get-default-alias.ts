import { Networks } from "@payvo/sdk";
import { Contracts } from "@payvo/sdk-profiles";

interface GetDefaultAliasInput {
	profile: Contracts.IProfile;
	network: Networks.Network;
}

export const getDefaultAlias = ({ profile, network }: GetDefaultAliasInput): string => {
	const makeAlias = (count: number) => `${network.displayName()} #${count}`;

	const sameCoinWallets = profile.wallets().findByCoinWithNetwork(network.coin(), network.id());

	let sameCoinWalletsCount = sameCoinWallets.length || 1;

	while (profile.wallets().findByAlias(makeAlias(sameCoinWalletsCount))) {
		sameCoinWalletsCount++;
	}

	return makeAlias(sameCoinWalletsCount);
};
