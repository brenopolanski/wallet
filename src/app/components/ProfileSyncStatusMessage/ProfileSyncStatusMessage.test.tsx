import React from "react";
import { Route } from "react-router-dom";
import { fireEvent, render, waitFor } from "utils/testing-library";

import { SyncErrorMessage } from "./ProfileSyncStatusMessage";

describe("SyncErrorMessage", () => {
	const failedNetworkNames = ["ARK Devnet", "ARK Mainnet", "Lisk Devnet"];

	it("should render one failed network", async () => {
		const { container, findByText } = render(
			<Route path="/">
				<SyncErrorMessage failedNetworkNames={[failedNetworkNames[0]]} />
			</Route>,
			{
				routes: ["/"],
			},
		);

		await findByText(failedNetworkNames[0]);

		expect(container).toMatchSnapshot();
	});

	it("should render two failed networks", async () => {
		const { container, findByText } = render(
			<Route path="/">
				<SyncErrorMessage failedNetworkNames={[failedNetworkNames[0], failedNetworkNames[1]]} />
			</Route>,
			{
				routes: ["/"],
			},
		);

		await findByText(failedNetworkNames[0]);
		await findByText(failedNetworkNames[1]);

		expect(container).toMatchSnapshot();
	});

	it("should render multiple failed networks", async () => {
		const { container, findByText } = render(
			<Route path="/">
				<SyncErrorMessage failedNetworkNames={failedNetworkNames} />
			</Route>,
			{
				routes: ["/"],
			},
		);

		await findByText(failedNetworkNames[0]);
		await findByText(failedNetworkNames[1]);

		expect(container).toMatchSnapshot();
	});

	it("should handle retry", async () => {
		const onRetry = jest.fn();
		const { container, getByTestId, findByText } = render(
			<Route path="/">
				<SyncErrorMessage failedNetworkNames={failedNetworkNames} onRetry={onRetry} />
			</Route>,
			{
				routes: ["/"],
			},
		);

		await findByText(failedNetworkNames[0]);
		await findByText(failedNetworkNames[1]);
		await findByText(failedNetworkNames[2]);

		fireEvent.click(getByTestId("SyncErrorMessage__retry"));

		await waitFor(() => expect(onRetry).toHaveBeenCalledWith());

		expect(container).toMatchSnapshot();
	});
});
