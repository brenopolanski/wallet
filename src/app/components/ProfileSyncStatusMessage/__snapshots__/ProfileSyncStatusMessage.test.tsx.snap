// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SyncErrorMessage should handle retry 1`] = `
<div>
  There has been a network problem on 
  <span>
    <strong>
      ARK Devnet
    </strong>
    , 
  </span>
  <span>
    <strong>
      ARK Mainnet
    </strong>
    , 
  </span>
  <span>
    <span>
       
      and
       
    </span>
    <strong>
      Lisk Devnet
    </strong>
  </span>
   while trying to retrieve your wallet information and as a result, your data could be out of sync or incorrect. Click 
  <a
    class="Link__AnchorStyled-sc-ec014t-0 eeKUfa group ring-focus"
    data-ring-focus-margin="-m-1"
    data-testid="SyncErrorMessage__retry"
    href="/"
  >
    <span
      class="Link__Content-sc-ec014t-1 fPOlTq"
    >
      here
    </span>
  </a>
   to retry.
</div>
`;

exports[`SyncErrorMessage should render multiple failed networks 1`] = `
<div>
  There has been a network problem on 
  <span>
    <strong>
      ARK Devnet
    </strong>
    , 
  </span>
  <span>
    <strong>
      ARK Mainnet
    </strong>
    , 
  </span>
  <span>
    <span>
       
      and
       
    </span>
    <strong>
      Lisk Devnet
    </strong>
  </span>
   while trying to retrieve your wallet information and as a result, your data could be out of sync or incorrect. Click 
  <a
    class="Link__AnchorStyled-sc-ec014t-0 eeKUfa group ring-focus"
    data-ring-focus-margin="-m-1"
    data-testid="SyncErrorMessage__retry"
    href="/"
  >
    <span
      class="Link__Content-sc-ec014t-1 fPOlTq"
    >
      here
    </span>
  </a>
   to retry.
</div>
`;

exports[`SyncErrorMessage should render one failed network 1`] = `
<div>
  There has been a network problem on 
  <span>
    <strong>
      ARK Devnet
    </strong>
  </span>
   while trying to retrieve your wallet information and as a result, your data could be out of sync or incorrect. Click 
  <a
    class="Link__AnchorStyled-sc-ec014t-0 eeKUfa group ring-focus"
    data-ring-focus-margin="-m-1"
    data-testid="SyncErrorMessage__retry"
    href="/"
  >
    <span
      class="Link__Content-sc-ec014t-1 fPOlTq"
    >
      here
    </span>
  </a>
   to retry.
</div>
`;

exports[`SyncErrorMessage should render two failed networks 1`] = `
<div>
  There has been a network problem on 
  <span>
    <strong>
      ARK Devnet
    </strong>
    , 
  </span>
  <span>
    <span>
       
      and
       
    </span>
    <strong>
      ARK Mainnet
    </strong>
  </span>
   while trying to retrieve your wallet information and as a result, your data could be out of sync or incorrect. Click 
  <a
    class="Link__AnchorStyled-sc-ec014t-0 eeKUfa group ring-focus"
    data-ring-focus-margin="-m-1"
    data-testid="SyncErrorMessage__retry"
    href="/"
  >
    <span
      class="Link__Content-sc-ec014t-1 fPOlTq"
    >
      here
    </span>
  </a>
   to retry.
</div>
`;
